narwhal/primary/src/core.rs

```diff
@@ -552,10 +560,18 @@ impl Core {
                 received: header.epoch
             }
         );
+        // Even if a certificate can be created from this header, it will never get included
```

Comment:
Certificates that we receive back from the certificate_waiter really should start their processing here? Because we know we have all parents. Might be worth splitting this function in 2 and calling the part below only when draining the loopback channel.

Reply:
Sanitize header is only used for header now.